python/cudf/cudf/core/column/column.py

```diff
@@ -1229,6 +1229,11 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             arbitrary = cupy.asarray(arbitrary).astype(arb_dtype)
             current_dtype = arb_dtype
 
+        if desc["strides"] is not None and desc.get("mask", None) is None:
```

Comment:
While `strides is None` is certainly one case that works, we should also check to see if `strides == (arb_dtype.itemsize,)` as that is also 1-D contiguous.

Reply:
Did you mean mean `if strides != (arb_dtype.itemsize,)` to check for 1-d non-contiguous?

I was not able to get a 1-d contiguous or non-contiguous array with such a condition, could you share an example?

```python
>>> cp.random.randn(10, 2)
array([[ 0.94513535,  0.83474103],
       [-0.48733383, -1.26725109],
       [ 0.92763362, -0.90840477],
       [ 1.51202585, -0.08442038],
       [-1.03553928,  2.14929072],
       [-0.64543545, -0.45847234],
       [-2.08376522, -0.82124291],
       [-0.72985977, -2.16106965],
       [ 0.88894551,  1.12507506],
       [-0.47210113, -1.03509318]])
>>> cp.random.randn(10, 2)[:, 0]
array([-1.40171207, -0.18492909, -0.67132592, -2.40527545,  1.49884205,
        2.01873371,  1.656129  , -1.00027593, -1.51870164,  0.17271589])
>>> cp.random.randn(10, 2)[:, 0].__cuda_array_interface__
{'shape': (10,), 'typestr': '<f8', 'descr': [('', '<f8')], 'version': 2, 'strides': (16,), 'data': (140367684960256, False)}
>>> np.dtype(cp.random.randn(10, 2)[:, 0].__cuda_array_interface__['typestr'])
dtype('float64')
>>> np.dtype(cp.random.randn(10, 2)[:, 0].__cuda_array_interface__['typestr']).itemsize
8
>>> cp.array([1, 2, 3])
array([1, 2, 3])
>>> cp.array([1, 2, 3]).__cuda_array_interface__
{'shape': (3,), 'typestr': '<i8', 'descr': [('', '<i8')], 'version': 2, 'strides': None, 'data': (140367684960256, False)}
>>> np.dtype(cp.array([1, 2, 3]).__cuda_array_interface__['typestr'])
dtype('int64')
>>> np.dtype(cp.array([1, 2, 3]).__cuda_array_interface__['typestr']).itemsize
8
```

Reply:
+1 to supporting where `strides == dtype.itemsize`

Reply:
Sure, how about this?

```python
In [1]: import cupy                                                             

In [2]: a = cupy.arange(5)                                                      

In [3]: a.itemsize, a.strides                                                   
Out[3]: (8, (8,))
```

Reply:
Thanks @jakirkham, this is now being handled.